pkg/orchestrator/discovery/discoverer.go

```diff
@@ -100,6 +101,10 @@ func (d *Discoverer) DiscoverAndCreateAssets(ctx context.Context) error {
 		// which matches the unique properties of this asset, in this
 		// case we'll patch the just AssetInfo and FirstSeen instead.
 		assetData.FirstSeen = nil
+		assetData.Providers = AppendProviderIfMissing(
+			conflictError.ConflictingAsset.Providers,
+			models.ProviderRelationship{Id: (*d.providerClient).Object().Id},
+		)
```

Comment:
I wonder if we need to handle the reverse case, we need to remove the provider ID from the asset if we do not find it. Alternatively we need to have a "Last Seen" field for each provider on the asset like:

```
{
   "id": "<uuid>",
   "assetInfo": {},
   "providers": [
        {
            "provider": { "id": "<uuid>" },
            "lastSeen": "<time stamp>"
        }
   ]
}
```

this would remove the need for the terminated field. In the orchestrator/scheduler we'd only schedule AssetScans onto providers which have seen the asset in the last 5 / 10 minutes or something like that.